### ISSUE TITLE
c2hs doesn't build under 32bit 

### ISSUE THREAD

**Issue by j-keck** [NixOS/nixpkgs#6343]
```
nix-env -iA nixos.pkgs.haskellngPackages.c2hs
installing ‘c2hs-0.22.1’
...
         Test Cases   Total       
 Passed  41           41          
 Failed  2            2           
 Total   43           43          
Test suite test-bugs: FAIL
Test suite logged to: dist/test/c2hs-0.22.1-test-bugs.log
Test suite test-system: RUNNING...
Test suite test-system: PASS
Test suite logged to: dist/test/c2hs-0.22.1-test-system.log
1 of 2 test suites (1 of 2 test cases) passed.
builder for ‘/nix/store/v3218rmf7qb1s0afl5hb4008b456i0aa-c2hs-0.22.1.drv’ failed with exit code 1
error: build of ‘/nix/store/v3218rmf7qb1s0afl5hb4008b456i0aa-c2hs-0.22.1.drv’ failed
```
System: Linux nixos 3.14.29 #1-NixOS SMP Thu Jan 1 00:00:01 UTC 1970 i686 GNU/Linux

upstream issue: haskell/c2hs#121, upstream pr: haskell/c2hs#122

**Comment by j-keck** [comment]
tests upstream fixed, new release scheduled for next week.

**Comment by peti** [comment]
Work-around committed in 81082af.

**Comment by j-keck** [comment]
upstream has a new version with the fix released (0.23.1).
@peti: should the work-around be removed when the new version is in nixpkgs?

**Comment by peti** [comment]
I'll remove the work-around in 2c9425b4b5aabcb7194f019cb3f437f31caee502. Thanks for reminding me!
